### PR TITLE
update float range to Hexadecimal Floating-Point Constants

### PR DESCRIPTION
**Pull request by dericed**

### CODE REVIEW

**Comment by mbunkus** [comment]
This pull request has conflicts due to me merging the colour property spec additions. Can  you please re-base it? Thanks.

File: spectool/specdata.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
   <element name="TimecodeScale" level="2" id="0x2AD7B1" type="uinteger" mandatory="1" minver="1" default="1000000">Timestamp scale in nanoseconds (1.000.000 means all timestamps in the segment are expressed in milliseconds).</element>
   <!-- <element name="TimecodeScaleDenominator" level="2" id="0x2AD7B2" type="uinteger" mandatory="1" minver="4" default="1000000000">Timestamp scale numerator, see <a href="http://www.matroska.org/technical/specs/index.html#TimecodeScale">TimecodeScale</a>.</element>
   TimecodeScale When combined with <a href="http://www.matroska.org/technical/specs/index.html#TimecodeScaleDenominator">TimecodeScaleDenominator</a> the Timestamp scale is given by the fraction TimecodeScale/TimecodeScaleDenominator in seconds.-->
-  <element name="Duration" level="2" id="0x4489" type="float" minver="1" range="&gt; 0">Duration of the segment (based on TimecodeScale).</element>
+  <element name="Duration" level="2" id="0x4489" type="float" minver="1" range="&gt;0x0p+1">Duration of the segment (based on TimecodeScale).</element>
```
Comment (robUx4) [inline spectool/specdata.xml]: needs spacing after &gt; for clarity. Also 0x0p+0 is a clearer representation of 0.0

File: spectool/specdata.xml [hunk 2]
```diff
@@ -88,7 +88,7 @@
...
   <element name="DefaultDecodedFieldDuration" cppname="TrackDefaultDecodedFieldDuration" level="3" id="0x234E7A" type="uinteger" minver="4" range="not 0">The period in nanoseconds (not scaled by TimcodeScale)
 between two successive fields at the output of the decoding process (see <a href="http://www.matroska.org/technical/specs/notes.html#DefaultDecodedFieldDuration">the notes</a>)</element>
-  <element name="TrackTimecodeScale" level="3" id="0x23314F" type="float" mandatory="1" minver="1" maxver="3" webm="0" default="1.0" range="&gt; 0">DEPRECATED, DO NOT USE. The scale to apply on this track to work at normal speed in relation with other tracks (mostly used to adjust video speed when the audio length differs).</element>
+  <element name="TrackTimecodeScale" level="3" id="0x23314F" type="float" mandatory="1" minver="1" maxver="3" webm="0" default="1.0" range="&gt;0x0p+1">DEPRECATED, DO NOT USE. The scale to apply on this track to work at normal speed in relation with other tracks (mostly used to adjust video speed when the audio length differs).</element>
```
Comment (robUx4) [inline spectool/specdata.xml]: same comment as above

File: spectool/specdata.xml [hunk 3]
```diff
@@ -123,11 +123,11 @@ between two successive fields at the output of the decoding process (see <a href
...
   <element name="AspectRatioType" cppname="VideoAspectRatio" level="4" id="0x54B3" type="uinteger" minver="1" default="0">Specify the possible modifications to the aspect ratio (0: free resizing, 1: keep aspect ratio, 2: fixed).</element>
   <element name="ColourSpace" cppname="VideoColourSpace" level="4" id="0x2EB524" type="binary" minver="1" webm="0" bytesize="4">Same value as in AVI (32 bits).</element>
-  <element name="GammaValue" cppname="VideoGamma" level="4" id="0x2FB523" type="float" webm="0" range="&gt; 0">Gamma Value.</element>
-  <element name="FrameRate" cppname="VideoFrameRate" level="4" id="0x2383E3" type="float" range="&gt; 0">Number of frames per second. <strong>Informational</strong> only.</element>
+  <element name="GammaValue" cppname="VideoGamma" level="4" id="0x2FB523" type="float" webm="0" range="&gt;0x0p+1">Gamma Value.</element>
+  <element name="FrameRate" cppname="VideoFrameRate" level="4" id="0x2383E3" type="float" range="&gt;0x0p+1">Number of frames per second. <strong>Informational</strong> only.</element>
```
Comment (robUx4) [inline spectool/specdata.xml]: idem

File: spectool/specdata.xml [hunk 4]
```diff
@@ -123,11 +123,11 @@ between two successive fields at the output of the decoding process (see <a href
   <element name="DisplayUnit" cppname="VideoDisplayUnit" level="4" id="0x54B2" type="uinteger" minver="1" default="0">How DisplayWidth &amp; DisplayHeight should be interpreted (0: pixels, 1: centimeters, 2: inches, 3: Display Aspect Ratio).</element>
   <element name="AspectRatioType" cppname="VideoAspectRatio" level="4" id="0x54B3" type="uinteger" minver="1" default="0">Specify the possible modifications to the aspect ratio (0: free resizing, 1: keep aspect ratio, 2: fixed).</element>
   <element name="ColourSpace" cppname="VideoColourSpace" level="4" id="0x2EB524" type="binary" minver="1" webm="0" bytesize="4">Same value as in AVI (32 bits).</element>
-  <element name="GammaValue" cppname="VideoGamma" level="4" id="0x2FB523" type="float" webm="0" range="&gt; 0">Gamma Value.</element>
-  <element name="FrameRate" cppname="VideoFrameRate" level="4" id="0x2383E3" type="float" range="&gt; 0">Number of frames per second. <strong>Informational</strong> only.</element>
+  <element name="GammaValue" cppname="VideoGamma" level="4" id="0x2FB523" type="float" webm="0" range="&gt;0x0p+1">Gamma Value.</element>
```
Comment (robUx4) [inline spectool/specdata.xml]: idem

File: spectool/specdata.xml [hunk 5]
```diff
@@ -123,11 +123,11 @@ between two successive fields at the output of the decoding process (see <a href
...
+  <element name="FrameRate" cppname="VideoFrameRate" level="4" id="0x2383E3" type="float" range="&gt;0x0p+1">Number of frames per second. <strong>Informational</strong> only.</element>
   <element name="Audio" cppname="TrackAudio" level="3" id="0xE1" type="master" minver="1">Audio settings.</element>
-  <element name="SamplingFrequency" cppname="AudioSamplingFreq" level="4" id="0xB5" type="float" mandatory="1" minver="1" default="8000.0" range="&gt; 0">Sampling frequency in Hz.</element>
-  <element name="OutputSamplingFrequency" cppname="AudioOutputSamplingFreq" level="4" id="0x78B5" type="float" minver="1" default="SamplingFrequency" range="&gt; 0">Real output sampling frequency in Hz (used for SBR techniques).</element>
+  <element name="SamplingFrequency" cppname="AudioSamplingFreq" level="4" id="0xB5" type="float" mandatory="1" minver="1" default="8000.0" range="&gt;0x0p+1">Sampling frequency in Hz.</element>
```
Comment (robUx4) [inline spectool/specdata.xml]: idem

File: spectool/specdata.xml [hunk 6]
```diff
@@ -123,11 +123,11 @@ between two successive fields at the output of the decoding process (see <a href
...
+  <element name="OutputSamplingFrequency" cppname="AudioOutputSamplingFreq" level="4" id="0x78B5" type="float" minver="1" default="SamplingFrequency" range="&gt;0x0p+1">Real output sampling frequency in Hz (used for SBR techniques).</element>
```
Comment (robUx4) [inline spectool/specdata.xml]: idem

File: spectool/specdata.xml [hunk 7]
```diff
@@ -88,7 +88,7 @@
...
   <element name="DefaultDecodedFieldDuration" cppname="TrackDefaultDecodedFieldDuration" level="3" id="0x234E7A" type="uinteger" minver="4" range="not 0">The period in nanoseconds (not scaled by TimcodeScale)
 between two successive fields at the output of the decoding process (see <a href="http://www.matroska.org/technical/specs/notes.html#DefaultDecodedFieldDuration">the notes</a>)</element>
-  <element name="TrackTimecodeScale" level="3" id="0x23314F" type="float" mandatory="1" minver="1" maxver="3" webm="0" default="1.0" range="&gt; 0">DEPRECATED, DO NOT USE. The scale to apply on this track to work at normal speed in relation with other tracks (mostly used to adjust video speed when the audio length differs).</element>
+  <element name="TrackTimecodeScale" level="3" id="0x23314F" type="float" mandatory="1" minver="1" maxver="3" webm="0" default="1.0" range="&gt; 0x0p+0">DEPRECATED, DO NOT USE. The scale to apply on this track to work at normal speed in relation with other tracks (mostly used to adjust video speed when the audio length differs).</element>
```
Comment (kit90) [inline spectool/specdata.xml]: `default="1.0"` should be `default="0x1p+0"` in hexadecimal notation.

File: spectool/specdata.xml [hunk 8]
```diff
@@ -123,11 +123,11 @@ between two successive fields at the output of the decoding process (see <a href
...
+  <element name="GammaValue" cppname="VideoGamma" level="4" id="0x2FB523" type="float" webm="0" range="&gt; 0x0p+0">Gamma Value.</element>
+  <element name="FrameRate" cppname="VideoFrameRate" level="4" id="0x2383E3" type="float" range="&gt; 0x0p+0">Number of frames per second. <strong>Informational</strong> only.</element>
   <element name="Audio" cppname="TrackAudio" level="3" id="0xE1" type="master" minver="1">Audio settings.</element>
-  <element name="SamplingFrequency" cppname="AudioSamplingFreq" level="4" id="0xB5" type="float" mandatory="1" minver="1" default="8000.0" range="&gt; 0">Sampling frequency in Hz.</element>
-  <element name="OutputSamplingFrequency" cppname="AudioOutputSamplingFreq" level="4" id="0x78B5" type="float" minver="1" default="SamplingFrequency" range="&gt; 0">Real output sampling frequency in Hz (used for SBR techniques).</element>
+  <element name="SamplingFrequency" cppname="AudioSamplingFreq" level="4" id="0xB5" type="float" mandatory="1" minver="1" default="8000.0" range="&gt; 0x0p+0">Sampling frequency in Hz.</element>
```
Comment (kit90) [inline spectool/specdata.xml]: 8000.0 = 0x1.f4p+12

File: spectool/specdata.xml (hunk 7 above)
Comment (dericed) [inline spectool/specdata.xml]: updated and squashed in https://github.com/Matroska-Org/foundation-source/pull/16/commits/28a63aa9c1fc7efa2e09950bb0a5c9593d7c5b21

File: spectool/specdata.xml (hunk 8 above)
Comment (dericed) [inline spectool/specdata.xml]: updated and squashed in https://github.com/Matroska-Org/foundation-source/pull/16/commits/28a63aa9c1fc7efa2e09950bb0a5c9593d7c5b21

**Comment by dericed** [comment]
@mbunkus conflicts are resolved.

**Comment by mbunkus** [comment]
Thank you.

**Pull request merged**